filebeat/input/netmetrics/udp.go

```diff
@@ -54,13 +52,12 @@ type UDP struct {
 }
 
 // NewUDP returns a new UDP input metricset. Note that if the id is empty then a nil UDP metricset is returned.
-func NewUDP(inputName string, id string, device string, buflen uint64, poll time.Duration, log *logp.Logger) *UDP {
+func NewUDP(id string, reg *monitoring.Registry, device string, buflen uint64, poll time.Duration, log *logp.Logger) *UDP {
```

Comment:
Here and elsewhere (there are a few other instances), where the `id` parameter is unused except for an empty-string check to abort the call, please remove the parameter and check for empty id at the call site instead.

Reply:
Whether the function returns a non-nil metrics object is not an "implementation detail," It is a documented invariant of the public API, and the id string now has no other effect.

This is not a request for a "refactor," there isn't even a real need to change the call site logic because the only consequence of a metrics object with an empty id is to track basic stats without reporting them to internal monitoring. You could do this by deleting eight lines.